src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -237,7 +237,28 @@ impl<'top> LazyRawBinaryValue_1_1<'top> {
 
     /// Helper method called by [`Self::read`]. Reads the current value as a float.
     fn read_float(&self) -> ValueParseResult<'top, BinaryEncoding_1_1> {
-        todo!();
+        debug_assert!(self.encoded_value.ion_type() == IonType::Float);
+
+        let value = match self.encoded_value.value_body_length {
+            8 => {
+                let mut buffer = [0; 8];
+                let val_bytes = self.input.slice(1, 8);
+                buffer[..8].copy_from_slice(val_bytes.bytes());
```

Comment:
Here and below, `ImmutableBuffer` has a method called `bytes_range` (that I really should have named `slice_bytes`) that skips making an intermediate `ImmutableBuffer`.
```suggestion
                let val_bytes = self.input.bytes_range(1, 8);
                buffer[..8].copy_from_slice(val_bytes);
```

Reply:
Oh awesome, I will add this change in, ty!